c++/cmake/CapnProtoTargets.cmake

```diff
@@ -0,0 +1,206 @@
+# This CMake script adds imported targets for each shared library and executable distributed by
+# Cap'n Proto's autotools build.
+#
+# This file IS NOT USED by the CMake build! The CMake build generates its own version of this script
+# from its set of exported targets. I used such a generated script as a reference when writing this
+# one.
+#
+# The set of library targets provided by this script is automatically generated from the list of .pc
+# files maintained in configure.ac. The set of executable targets is hard-coded in this file.
+#
+# You can request this script print debugging information by invoking cmake with:
+#
+#   -DCapnProto_DEBUG=ON
+#
+# TODO(someday): Distinguish between debug and release builds. I.e., set IMPORTED_LOCATION_RELEASE
+#   rather than IMPORTED_LOCATION, etc., if this installation was configured as a release build. But
+#   how do we tell? grep for -g in CXXFLAGS?
+
+if(CMAKE_VERSION VERSION_LESS 3.1)
+  message(FATAL_ERROR "CMake >= 3.1 required")
+endif()
+
+set(forwarded_config_flags)
+if(CapnProto_FIND_QUIETLY)
+  list(APPEND forwarded_config_flags QUIET)
+endif()
+if(CapnProto_FIND_REQUIRED)
+  list(APPEND forwarded_config_flags REQUIRED)
+endif()
+# If the consuming project called find_package(CapnProto) with the QUIET or REQUIRED flags, forward
+# them to calls to find_package(PkgConfig) and pkg_check_modules(). Note that find_dependency()
+# would do this for us in the former case, but there is no such forwarding wrapper for
+# pkg_check_modules().
+
+find_package(PkgConfig ${forwarded_config_flags})
+if(NOT ${PkgConfig_FOUND})
+  # If we're here, the REQUIRED flag must not have been passed, else we would have had a fatal
+  # error. Nevertheless, a diagnostic for this case is probably nice.
+  message(WARNING "pkg-config cannot be found")
```

Comment:
Do not print warning if QUIET was [specified](https://itk.org/Wiki/CMake/Examples#Suppress_warnings)?

Reply:
Will fix.

I went back and forth on what the right answer here is -- not having pkg-config installed seems like a serious problem, orthogonal to whether or not Cap'n Proto is installed, so I wasn't sure whether that should still be suppressed by the QUIET flag or not. I'm happy suppressing it for consistency's sake, though.